crates/bevy_ecs/src/event.rs

```diff
@@ -187,7 +187,8 @@ impl<T> Default for ManualEventReader<T> {
     }
 }
 
-impl<T> ManualEventReader<T> {
+#[allow(clippy::len_without_is_empty)] // Check fails since the is_empty implementation has a signature other than `(&self) -> bool`
+impl<T: Resource> ManualEventReader<T> {
```

Comment:
Because it calls `events.event_reader_len()` which is only implemented for `Events<T: Resource>`. This seems reasonable to me.

Reply:
Although in practice, events dont need to be resources themselves. This could get more confusing if we decide to require Resource derives / enable more configuration options. Removing that bound in favor of something like (`Send + Sync`) is worth considering (in a separate issue / pr).